### PR TITLE
Module2 kyrtaq

### PR DESCRIPTION
**Pull request by kyrtaq**

### CODE REVIEW

File: module2/homework/transformContainers/test.cpp [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #include "gtest/gtest.h"
 
 // TODO: add proper includes
+#include <algorithm>
+
```
Comment (NG90) [inline module2/homework/transformContainers/test.cpp:5]: not needed new line

File: module2/homework/transformContainers/tranform.hpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+#include <deque>
+#include <list>
+#include <map>
+#include <string>
+
+std::map<int, std::string> removeDuplicateAndTranformToMap(std::list<std::string>, std::deque<int>);
```
Comment (NG90) [inline module2/homework/transformContainers/tranform.hpp]: Try pass arguments as const reference.

File: module2/homework/transformContainers/transform.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+std::map<int, std::string> removeDuplicateAndTranformToMap(
+    std::list<std::string> uniqueList,
+    std::deque<int> uniqueDeque) {
+    uniqueList.sort();
+    uniqueList.erase(std::unique(uniqueList.begin(), uniqueList.end()), uniqueList.end());
+
+    for (const auto el : uniqueDeque) {
+        std::cout << el << ' ';
+    }
```
Comment (NG90) [inline module2/homework/transformContainers/transform.cpp]: It looks as some debug print. Should be removed in final version

File: module2/homework/transformContainers/transform.cpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    std::sort(uniqueDeque.begin(), uniqueDeque.end());
+    uniqueDeque.erase(std::unique(uniqueDeque.begin(), uniqueDeque.end()), uniqueDeque.end());
+
+    std::map<int, std::string> result{};
+    std::transform(uniqueDeque.begin(),
+                   uniqueDeque.end(),
+                   uniqueList.begin(),
+                   std::inserter(result, result.end()),
```
Comment (NG90) [inline module2/homework/transformContainers/transform.cpp:24]: you can use:
`std::back_inserter(result)`

File: module2/homework/transformContainers/transform.cpp [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    for (const auto el : uniqueDeque) {
+        std::cout << el << ' ';
+    }
+
+    std::sort(uniqueDeque.begin(), uniqueDeque.end());
+    uniqueDeque.erase(std::unique(uniqueDeque.begin(), uniqueDeque.end()), uniqueDeque.end());
+
+    std::map<int, std::string> result{};
+    std::transform(uniqueDeque.begin(),
```
Comment (NG90) [inline module2/homework/transformContainers/transform.cpp:21]: I think before transform algorithm it should be checked, whether list and deque have the same amount of elements. If not, empty map could be returned.

File: module2/homework/transformContainers/test.cpp [hunk 6]
```diff
@@ -1,6 +1,9 @@
...
+#include "tranform.hpp"
 
 TEST(transformContainerTests, ShouldReturnUniqueMap) {
```
Comment (NG90) [inline module2/homework/transformContainers/test.cpp:8]: Please think about adding one more test, where list and deque without duplicates have different amount of elements.
IMO in such case function should just return empty map.

**Review by NG90 — CHANGES_REQUESTED** [review]

File: module2/homework/transformContainers/test.cpp (hunk 1 above)
Comment (kyrtaq) [reply module2/homework/transformContainers/test.cpp:5]: its from auto formatting

File: module2/homework/transformContainers/transform.cpp (hunk 4 above)
Comment (kyrtaq) [reply module2/homework/transformContainers/transform.cpp:24]: it doesnt compile

Comment (NG90) [reply module2/homework/transformContainers/transform.cpp:24]: you're right. back_inserter doesn't work for string, sorry

File: module2/homework/transformContainers/transform.cpp [hunk 7]
```diff
@@ -0,0 +1,29 @@
+#include <algorithm>
+#include <iostream>
+
+#include "tranform.hpp"
+
+std::map<int, std::string> removeDuplicateAndTranformToMap(
+    std::list<std::string>& uniqueList,
+    std::deque<int>& uniqueDeque) {
```
Comment (NG90) [inline module2/homework/transformContainers/transform.cpp]: I know I adviced you to use reference here, but I re-think this advice and it was wrong idea.
In current solution, the original input conteiners,  list and deque are modified, during e.g. sorting. I think this function should not modify original vakues, so passing by copy is better here.
Sorry for that.

**Review by NG90 — APPROVED** [review]
